src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -309,7 +310,10 @@ public bool Remove(T item)
                             _slots[last].next = _slots[i].next;
                         }
                         _slots[i].hashCode = -1;
-                        _slots[i].value = default(T);
+                        if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
```

Comment:
It's an intrinsic and shouldn't even show up in the asm. (This code path is also only run when we're about to return from the method, having found the thing we're looking for.)